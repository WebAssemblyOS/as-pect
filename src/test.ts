/**
 * This file is used by the package.json "asp" script to simulate using the command line interface.
 */
require("../lib/cli/index.js").asp(process.argv.slice(2));
