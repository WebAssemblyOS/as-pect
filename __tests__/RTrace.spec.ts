import { TestContext } from "../src/test/TestContext";
import { createRTraceModule } from "./setup/createRTraceModule";

let ctx: TestContext;

let start = new Promise<void>((resolve, reject) => {
  createRTraceModule({}, (err, result) => {
    if (err) {
      console.log(err);
      reject(err);
    } else {
      ctx = result!;
      resolve();
    }
  });
});

beforeEach(() => start);

describe("RTrace output", () => {
  test("RTrace group and test values", () => {
    expect(ctx.freeCount).toMatchSnapshot("freeCount");
    expect(ctx.allocationCount).toMatchSnapshot("allocationCount");
    expect(ctx.errors).toMatchSnapshot("errors");

    for (let i = 0; i < ctx.testGroups.length; i++) {
      const group = ctx.testGroups[i];
      expect(group.freeCount).toMatchSnapshot(`group ${i} freeCount`);
      expect(group.allocationCount).toMatchSnapshot(`group ${i} allocationCount`);
      expect(group.incrementCount).toMatchSnapshot(`group ${i} incrementCount`);
      expect(group.decrementCount).toMatchSnapshot(`group ${i} decrementCount`);
      expect(group.rtraceStart).toMatchSnapshot(`group ${i} rtraceStart`);
      expect(group.rtraceEnd).toMatchSnapshot(`group ${i} rtraceEnd`);
      expect(group.rtraceDelta).toMatchSnapshot(`group ${i} rtraceDelta`);
      expect(group.errors).toMatchSnapshot(`group ${i} errors`);

      for (let j = 0; j < group.tests.length; j++) {
        const test = group.tests[j];
        expect(test.freeCount).toMatchSnapshot(`test ${i}-${j} freeCount`);
        expect(test.allocationCount).toMatchSnapshot(`test ${i}-${j} allocationCount`);
        expect(test.incrementCount).toMatchSnapshot(`test ${i}-${j} incrementCount`);
        expect(test.decrementCount).toMatchSnapshot(`test ${i}-${j} decrementCount`);
        expect(test.rtraceStart).toMatchSnapshot(`test ${i}-${j} rtraceStart`);
        expect(test.rtraceEnd).toMatchSnapshot(`test ${i}-${j} rtraceEnd`);
        expect(test.rtraceDelta).toMatchSnapshot(`test ${i}-${j} rtraceDelta`);
        expect(test.errors).toMatchSnapshot(`test ${i}-${j} errors`);
      }
    }
  });
});