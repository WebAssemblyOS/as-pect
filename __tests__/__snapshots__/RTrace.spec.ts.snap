// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RTrace output RTrace group and test values: allocationCount 1`] = `16`;

exports[`RTrace output RTrace group and test values: errors 1`] = `Array []`;

exports[`RTrace output RTrace group and test values: freeCount 1`] = `4`;

exports[`RTrace output RTrace group and test values: group 0 allocationCount 1`] = `8`;

exports[`RTrace output RTrace group and test values: group 0 decrementCount 1`] = `8`;

exports[`RTrace output RTrace group and test values: group 0 errors 1`] = `Array []`;

exports[`RTrace output RTrace group and test values: group 0 freeCount 1`] = `2`;

exports[`RTrace output RTrace group and test values: group 0 incrementCount 1`] = `8`;

exports[`RTrace output RTrace group and test values: group 0 rtraceDelta 1`] = `6`;

exports[`RTrace output RTrace group and test values: group 0 rtraceEnd 1`] = `9`;

exports[`RTrace output RTrace group and test values: group 0 rtraceStart 1`] = `3`;

exports[`RTrace output RTrace group and test values: group 1 allocationCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: group 1 decrementCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: group 1 errors 1`] = `Array []`;

exports[`RTrace output RTrace group and test values: group 1 freeCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: group 1 incrementCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: group 1 rtraceDelta 1`] = `0`;

exports[`RTrace output RTrace group and test values: group 1 rtraceEnd 1`] = `0`;

exports[`RTrace output RTrace group and test values: group 1 rtraceStart 1`] = `9`;

exports[`RTrace output RTrace group and test values: group 2 allocationCount 1`] = `5`;

exports[`RTrace output RTrace group and test values: group 2 decrementCount 1`] = `13`;

exports[`RTrace output RTrace group and test values: group 2 errors 1`] = `Array []`;

exports[`RTrace output RTrace group and test values: group 2 freeCount 1`] = `1`;

exports[`RTrace output RTrace group and test values: group 2 incrementCount 1`] = `17`;

exports[`RTrace output RTrace group and test values: group 2 rtraceDelta 1`] = `4`;

exports[`RTrace output RTrace group and test values: group 2 rtraceEnd 1`] = `12`;

exports[`RTrace output RTrace group and test values: group 2 rtraceStart 1`] = `8`;

exports[`RTrace output RTrace group and test values: test 0-0 allocationCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-0 decrementCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-0 errors 1`] = `Array []`;

exports[`RTrace output RTrace group and test values: test 0-0 freeCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-0 incrementCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-0 rtraceDelta 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-0 rtraceEnd 1`] = `3`;

exports[`RTrace output RTrace group and test values: test 0-0 rtraceStart 1`] = `3`;

exports[`RTrace output RTrace group and test values: test 0-1 allocationCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-1 decrementCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-1 errors 1`] = `Array []`;

exports[`RTrace output RTrace group and test values: test 0-1 freeCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-1 incrementCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-1 rtraceDelta 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-1 rtraceEnd 1`] = `3`;

exports[`RTrace output RTrace group and test values: test 0-1 rtraceStart 1`] = `3`;

exports[`RTrace output RTrace group and test values: test 0-2 allocationCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-2 decrementCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-2 errors 1`] = `Array []`;

exports[`RTrace output RTrace group and test values: test 0-2 freeCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-2 incrementCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-2 rtraceDelta 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-2 rtraceEnd 1`] = `3`;

exports[`RTrace output RTrace group and test values: test 0-2 rtraceStart 1`] = `3`;

exports[`RTrace output RTrace group and test values: test 0-3 allocationCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-3 decrementCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-3 errors 1`] = `Array []`;

exports[`RTrace output RTrace group and test values: test 0-3 freeCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-3 incrementCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-3 rtraceDelta 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-3 rtraceEnd 1`] = `3`;

exports[`RTrace output RTrace group and test values: test 0-3 rtraceStart 1`] = `3`;

exports[`RTrace output RTrace group and test values: test 0-4 allocationCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-4 decrementCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-4 errors 1`] = `Array []`;

exports[`RTrace output RTrace group and test values: test 0-4 freeCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-4 incrementCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-4 rtraceDelta 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-4 rtraceEnd 1`] = `3`;

exports[`RTrace output RTrace group and test values: test 0-4 rtraceStart 1`] = `3`;

exports[`RTrace output RTrace group and test values: test 0-5 allocationCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-5 decrementCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-5 errors 1`] = `Array []`;

exports[`RTrace output RTrace group and test values: test 0-5 freeCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-5 incrementCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-5 rtraceDelta 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-5 rtraceEnd 1`] = `3`;

exports[`RTrace output RTrace group and test values: test 0-5 rtraceStart 1`] = `3`;

exports[`RTrace output RTrace group and test values: test 0-6 allocationCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-6 decrementCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-6 errors 1`] = `Array []`;

exports[`RTrace output RTrace group and test values: test 0-6 freeCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-6 incrementCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-6 rtraceDelta 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-6 rtraceEnd 1`] = `3`;

exports[`RTrace output RTrace group and test values: test 0-6 rtraceStart 1`] = `3`;

exports[`RTrace output RTrace group and test values: test 0-7 allocationCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-7 decrementCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-7 errors 1`] = `Array []`;

exports[`RTrace output RTrace group and test values: test 0-7 freeCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-7 incrementCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-7 rtraceDelta 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-7 rtraceEnd 1`] = `3`;

exports[`RTrace output RTrace group and test values: test 0-7 rtraceStart 1`] = `3`;

exports[`RTrace output RTrace group and test values: test 0-8 allocationCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-8 decrementCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-8 errors 1`] = `Array []`;

exports[`RTrace output RTrace group and test values: test 0-8 freeCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-8 incrementCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-8 rtraceDelta 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-8 rtraceEnd 1`] = `3`;

exports[`RTrace output RTrace group and test values: test 0-8 rtraceStart 1`] = `3`;

exports[`RTrace output RTrace group and test values: test 0-9 allocationCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-9 decrementCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-9 errors 1`] = `Array []`;

exports[`RTrace output RTrace group and test values: test 0-9 freeCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-9 incrementCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-9 rtraceDelta 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-9 rtraceEnd 1`] = `3`;

exports[`RTrace output RTrace group and test values: test 0-9 rtraceStart 1`] = `3`;

exports[`RTrace output RTrace group and test values: test 0-10 allocationCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-10 decrementCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-10 errors 1`] = `Array []`;

exports[`RTrace output RTrace group and test values: test 0-10 freeCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-10 incrementCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-10 rtraceDelta 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-10 rtraceEnd 1`] = `3`;

exports[`RTrace output RTrace group and test values: test 0-10 rtraceStart 1`] = `3`;

exports[`RTrace output RTrace group and test values: test 0-11 allocationCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-11 decrementCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-11 errors 1`] = `Array []`;

exports[`RTrace output RTrace group and test values: test 0-11 freeCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-11 incrementCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-11 rtraceDelta 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-11 rtraceEnd 1`] = `3`;

exports[`RTrace output RTrace group and test values: test 0-11 rtraceStart 1`] = `3`;

exports[`RTrace output RTrace group and test values: test 0-12 allocationCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-12 decrementCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-12 errors 1`] = `Array []`;

exports[`RTrace output RTrace group and test values: test 0-12 freeCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-12 incrementCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-12 rtraceDelta 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-12 rtraceEnd 1`] = `3`;

exports[`RTrace output RTrace group and test values: test 0-12 rtraceStart 1`] = `3`;

exports[`RTrace output RTrace group and test values: test 0-13 allocationCount 1`] = `2`;

exports[`RTrace output RTrace group and test values: test 0-13 decrementCount 1`] = `1`;

exports[`RTrace output RTrace group and test values: test 0-13 errors 1`] = `Array []`;

exports[`RTrace output RTrace group and test values: test 0-13 freeCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-13 incrementCount 1`] = `1`;

exports[`RTrace output RTrace group and test values: test 0-13 rtraceDelta 1`] = `2`;

exports[`RTrace output RTrace group and test values: test 0-13 rtraceEnd 1`] = `5`;

exports[`RTrace output RTrace group and test values: test 0-13 rtraceStart 1`] = `3`;

exports[`RTrace output RTrace group and test values: test 0-14 allocationCount 1`] = `4`;

exports[`RTrace output RTrace group and test values: test 0-14 decrementCount 1`] = `2`;

exports[`RTrace output RTrace group and test values: test 0-14 errors 1`] = `Array []`;

exports[`RTrace output RTrace group and test values: test 0-14 freeCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-14 incrementCount 1`] = `2`;

exports[`RTrace output RTrace group and test values: test 0-14 rtraceDelta 1`] = `2`;

exports[`RTrace output RTrace group and test values: test 0-14 rtraceEnd 1`] = `7`;

exports[`RTrace output RTrace group and test values: test 0-14 rtraceStart 1`] = `5`;

exports[`RTrace output RTrace group and test values: test 0-15 allocationCount 1`] = `6`;

exports[`RTrace output RTrace group and test values: test 0-15 decrementCount 1`] = `3`;

exports[`RTrace output RTrace group and test values: test 0-15 errors 1`] = `Array []`;

exports[`RTrace output RTrace group and test values: test 0-15 freeCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-15 incrementCount 1`] = `3`;

exports[`RTrace output RTrace group and test values: test 0-15 rtraceDelta 1`] = `2`;

exports[`RTrace output RTrace group and test values: test 0-15 rtraceEnd 1`] = `9`;

exports[`RTrace output RTrace group and test values: test 0-15 rtraceStart 1`] = `7`;

exports[`RTrace output RTrace group and test values: test 0-16 allocationCount 1`] = `7`;

exports[`RTrace output RTrace group and test values: test 0-16 decrementCount 1`] = `5`;

exports[`RTrace output RTrace group and test values: test 0-16 errors 1`] = `Array []`;

exports[`RTrace output RTrace group and test values: test 0-16 freeCount 1`] = `1`;

exports[`RTrace output RTrace group and test values: test 0-16 incrementCount 1`] = `5`;

exports[`RTrace output RTrace group and test values: test 0-16 rtraceDelta 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-16 rtraceEnd 1`] = `9`;

exports[`RTrace output RTrace group and test values: test 0-16 rtraceStart 1`] = `9`;

exports[`RTrace output RTrace group and test values: test 0-17 allocationCount 1`] = `8`;

exports[`RTrace output RTrace group and test values: test 0-17 decrementCount 1`] = `8`;

exports[`RTrace output RTrace group and test values: test 0-17 errors 1`] = `Array []`;

exports[`RTrace output RTrace group and test values: test 0-17 freeCount 1`] = `2`;

exports[`RTrace output RTrace group and test values: test 0-17 incrementCount 1`] = `8`;

exports[`RTrace output RTrace group and test values: test 0-17 rtraceDelta 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 0-17 rtraceEnd 1`] = `9`;

exports[`RTrace output RTrace group and test values: test 0-17 rtraceStart 1`] = `9`;

exports[`RTrace output RTrace group and test values: test 1-0 allocationCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 1-0 decrementCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 1-0 errors 1`] = `Array []`;

exports[`RTrace output RTrace group and test values: test 1-0 freeCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 1-0 incrementCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 1-0 rtraceDelta 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 1-0 rtraceEnd 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 1-0 rtraceStart 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 2-0 allocationCount 1`] = `1`;

exports[`RTrace output RTrace group and test values: test 2-0 decrementCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 2-0 errors 1`] = `Array []`;

exports[`RTrace output RTrace group and test values: test 2-0 freeCount 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 2-0 incrementCount 1`] = `1`;

exports[`RTrace output RTrace group and test values: test 2-0 rtraceDelta 1`] = `1`;

exports[`RTrace output RTrace group and test values: test 2-0 rtraceEnd 1`] = `9`;

exports[`RTrace output RTrace group and test values: test 2-0 rtraceStart 1`] = `8`;

exports[`RTrace output RTrace group and test values: test 2-1 allocationCount 1`] = `1`;

exports[`RTrace output RTrace group and test values: test 2-1 decrementCount 1`] = `1`;

exports[`RTrace output RTrace group and test values: test 2-1 errors 1`] = `Array []`;

exports[`RTrace output RTrace group and test values: test 2-1 freeCount 1`] = `1`;

exports[`RTrace output RTrace group and test values: test 2-1 incrementCount 1`] = `1`;

exports[`RTrace output RTrace group and test values: test 2-1 rtraceDelta 1`] = `-1`;

exports[`RTrace output RTrace group and test values: test 2-1 rtraceEnd 1`] = `8`;

exports[`RTrace output RTrace group and test values: test 2-1 rtraceStart 1`] = `9`;

exports[`RTrace output RTrace group and test values: test 2-2 allocationCount 1`] = `4`;

exports[`RTrace output RTrace group and test values: test 2-2 decrementCount 1`] = `7`;

exports[`RTrace output RTrace group and test values: test 2-2 errors 1`] = `Array []`;

exports[`RTrace output RTrace group and test values: test 2-2 freeCount 1`] = `1`;

exports[`RTrace output RTrace group and test values: test 2-2 incrementCount 1`] = `10`;

exports[`RTrace output RTrace group and test values: test 2-2 rtraceDelta 1`] = `3`;

exports[`RTrace output RTrace group and test values: test 2-2 rtraceEnd 1`] = `11`;

exports[`RTrace output RTrace group and test values: test 2-2 rtraceStart 1`] = `8`;

exports[`RTrace output RTrace group and test values: test 2-3 allocationCount 1`] = `4`;

exports[`RTrace output RTrace group and test values: test 2-3 decrementCount 1`] = `7`;

exports[`RTrace output RTrace group and test values: test 2-3 errors 1`] = `Array []`;

exports[`RTrace output RTrace group and test values: test 2-3 freeCount 1`] = `1`;

exports[`RTrace output RTrace group and test values: test 2-3 incrementCount 1`] = `10`;

exports[`RTrace output RTrace group and test values: test 2-3 rtraceDelta 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 2-3 rtraceEnd 1`] = `11`;

exports[`RTrace output RTrace group and test values: test 2-3 rtraceStart 1`] = `11`;

exports[`RTrace output RTrace group and test values: test 2-4 allocationCount 1`] = `5`;

exports[`RTrace output RTrace group and test values: test 2-4 decrementCount 1`] = `9`;

exports[`RTrace output RTrace group and test values: test 2-4 errors 1`] = `Array []`;

exports[`RTrace output RTrace group and test values: test 2-4 freeCount 1`] = `1`;

exports[`RTrace output RTrace group and test values: test 2-4 incrementCount 1`] = `13`;

exports[`RTrace output RTrace group and test values: test 2-4 rtraceDelta 1`] = `1`;

exports[`RTrace output RTrace group and test values: test 2-4 rtraceEnd 1`] = `12`;

exports[`RTrace output RTrace group and test values: test 2-4 rtraceStart 1`] = `11`;

exports[`RTrace output RTrace group and test values: test 2-5 allocationCount 1`] = `5`;

exports[`RTrace output RTrace group and test values: test 2-5 decrementCount 1`] = `13`;

exports[`RTrace output RTrace group and test values: test 2-5 errors 1`] = `Array []`;

exports[`RTrace output RTrace group and test values: test 2-5 freeCount 1`] = `1`;

exports[`RTrace output RTrace group and test values: test 2-5 incrementCount 1`] = `17`;

exports[`RTrace output RTrace group and test values: test 2-5 rtraceDelta 1`] = `0`;

exports[`RTrace output RTrace group and test values: test 2-5 rtraceEnd 1`] = `12`;

exports[`RTrace output RTrace group and test values: test 2-5 rtraceStart 1`] = `12`;
